CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+# Contributing
+
+Contributions are welcome! Here's how you can help:
+
+- [Contributing code](#code)
+- [Reporting issues](#issues)
+- [Requesting features](#feature-requests)
+- [Translating](#translations)
+- [Donating](#donations)
+
+## Code
+
+If you are planning to start some significant coding, you would benefit from asking first on [our IRC channel](http://www.minetest.net/irc/) before starting.
+
+1. [Fork](https://help.github.com/articles/fork-a-repo/) the repository and [clone](https://help.github.com/articles/cloning-a-repository/) your fork.
+
+2. Start coding!
+    - Refer to the [Lua API](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt), [Developer Wiki](http://dev.minetest.net/Main_Page) and other [documentation](https://github.com/minetest/minetest/tree/master/doc).
+    - Follow the [C/C++](http://dev.minetest.net/Code_style_guidelines) and [Lua](http://dev.minetest.net/Lua_code_style_guidelines) code style guidelines.
+    - Check your code works as expected and document any changes to the Lua API.
+
+3. Commit & [push](https://help.github.com/articles/pushing-to-a-remote/) your changes to a new branch (not `master`, one change per branch)
+    - Commit messages should:
+        - Use the present tense
+        - Have a title which begins with a capital letter
+        - Be descriptive. (e.g. no `Update init.lua` or `Fix a problem`)
+        - Have a first line with less than *80 characters* and have a second line that is *empty*
+
+4. Once you are happy with your changes, submit a pull request.
+     - Open the [pull-request form](https://github.com/minetest/minetest/pull/new/master)
+     - Add a short description explaining briefly what you've done (or if it's a work-in-progress - what you need to do)
+
+##### A pull-request is considered merge-able when:
+
+1. It follows the [roadmap](https://forum.minetest.net/viewtopic.php?t=9177) in some way and fits the whole picture of the project.
+2. It works.
+3. It follows the code style for [C/C++](http://dev.minetest.net/Code_style_guidelines) or [Lua](http://dev.minetest.net/Lua_code_style_guidelines).
+4. The code's internal interfaces are reasonably optimized, depending on how often the code is called.
```

Comment:
I was about to consider merging this, but... wait... what happened to this line? I don't think it makes much sense now. As I originally wrote it, it said something along the lines of "interfaces must be well designed", exactly because of the reason that the implementation can be optimized later, but if the interfaces are crappy, fixing it is way more work.

I mean... does anyone even agree with my original thought here then?